FirebaseAuth/Sources/Swift/MultiFactor/MultiFactorSession.swift

```diff
@@ -39,8 +39,13 @@ import Foundation
     /// Current user object.
     var currentUser: User?
 
-    class var sessionForCurrentUser: MultiFactorSession {
-      guard let currentUser = Auth.auth().currentUser else {
+    class func session(for user: User?) -> MultiFactorSession {
+      let currentUser = if user != nil {
+        user
+      } else {
+        Auth.auth().currentUser
+      }
```

Comment:
optional: seems like a good use case for the nil coalescing operator
```suggestion
      let currentUser = user ?? Auth.auth().currentUser
```